wordpress-mobile/WordPress-Android
# Issue/13329 choose items to recyclerview

Parent #13329 

This PR swaps in a recyclerView for the inflexible hardcoded layout to support selecting available items to include in the downloadable backup. At the time of this writing, the available items are not dynamic, so I put them into `BackupAvialableItemsProvider` so it can be reused with the Restore details view. This PR also introduces base UiStates for the BackupDownloadDetails view. The UiStates will continue to be built out in future PRs.

Notes:
- Unit Test are following in PR #13545 
- Design has not been finalized
- Checkboxes default to checked and will not persist across device rotation yet (see todo's)

**Merge instructions**
- Make sure ~~#13520~~ & #13532 have been merged
- Remove Not Ready For Merge label
- Merge as normal

**To test:**
- Enable the BackupFeatureConfig and restart the app
- Navigate to the Activity Log
- Tap on the More menu icon
- Tap the "Create downloadable backup" option
- The backup detail view is shown 
- Tap/untap on the checkboxes

PR submission checklist:
